data-raw/R2R_TMH_habitat_inputs/documentation/TMH_model_implementation_overview.Rmd

```diff
@@ -0,0 +1,260 @@
+```
+
+### Theoretical Maximum Habitat Inputs
+
+Habitat inputs for Reorienting to Recovery theoretical maximum habitat model run was scaled existing habitat to area curves using the 2018/2019 BiOp flows.
+
+-   `DSMhabitat::fr_spawn$biop_itp_2018_2019`
+-   `DSMhabitat::fr_juv$biop_itp_2018_2019`
+-   `DSMhabitat::fr_fry$biop_itp_2018_2019`
+-   `DSMhabitat::fr_fp$biop_itp_2018_2019`
+-   `DSMhabitat::fr_delta$biop_itp_2018_2019`
+
+For each river reach in the [SIT DSM]('https://cvpia-osc.github.io/DSMhabitat/articles/habitat-extents-map.html'), the theoretical maximum habitat was calculated for above and below dams and for unregulated rivers. See methodology (TODO LINK to methodology) for in-depth overview of theoretical maximum habitat modeling.
+
+Theoretical maximum habitat was not calculated at specific flows and therefore to combine this added habitat with existing (at baseline), the existing was scaled using the following approach.
+
+**1) Calculate flow point of reference**
+
+For in-channel habitat we use the median flow over the entire modeled period 1980 - 2000. Flow inputs come from CalSim II benchmark from the 2019 Itp and 2020 BiOp. For floodplain habitat, we use a 2 year flow value, which represents a flow that is exceeded in 50% of the years between 1980 and 2000. This is done by take the yearly rolling maximum of monthly CalSim flows and applying a cumulative distribution function on the ranked values to find the proportion of all values less than or equal to the ranked order. An interpolation function is then developed to to find the 2 year flow value.
+
+```{r include=FALSE}
+roll_stat = min
+annual_stat = max
+
+exceedance_probs_monthly <- function(data, roll_stat, annual_stat) {
+  annual_durations <- data %>%
+    mutate(water_year = ifelse(month(date) %in% 10:12, year(date) + 1, year(date))) 
+  
+  annual_stats <- annual_durations %>%
+    group_by(water_year) %>%
+    summarise(stat_in_duration = annual_stat(flow_cfs, na.rm = TRUE))
+  
+  annual_stats %>%
+    mutate(dist = round(cume_dist(-stat_in_duration), 3)) %>%
+    arrange(dist)
+  
+}
+
+calsim_30_day <- function(data, exceedance_function) {
+  dur_30_min_max <- exceedance_probs_monthly(data, roll_stat = roll_stat, annual_stat = annual_stat)
+  
+  interpolate_probs_30_min_max <- approxfun(x = dur_30_min_max$dist, 
+                                            dur_30_min_max$stat_in_duration)
+  
+  d30 <- interpolate_probs_30_min_max(0.5) 
+  
+  return(d30)
+}
+
+```
+
+```{r include=FALSE}
+watershed_input <- "Cosumnes River"
+
+flood_flow_ex = DSMflow::flows_cfs$biop_itp_2018_2019 |> 
+      filter(date >= as_date("1980-01-01")) |> 
+      select(watershed_input, date) |> 
+      rename(flow_cfs = watershed_input) 
+    
+exceedance_probs_monthly_fun <- exceedance_probs_monthly(flood_flow_ex, 'min', 'max')
+calsim_30_day(flood_flow_ex, exceedance_probs_monthly_fun)
+
+ggplot(exceedance_probs_monthly_fun, aes(x = dist, y = stat_in_duration)) + 
```

Comment:
please fix axis labels for this plot

Reply:
More of an exploratory plot that was not included in the output. I'll comment out.